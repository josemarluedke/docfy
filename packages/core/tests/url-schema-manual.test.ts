import Docfy from '../src';
import path from 'path';

const root = path.resolve(__dirname, './__fixtures__/monorepo');

describe('URLs should be correctly generated when urlSchema is set to manual', () => {
  test('base base, no url prefix, no suffix', async () => {
    const docfy = new Docfy();
    const result = await docfy.run([
      {
        root,
        urlSchema: 'manual',
        pattern: '**/*.md',
      },
    ]);

    const urls = [];
    result.content.forEach(page => {
      urls.push([page.source, page.meta.url]);
    });
    expect(urls).toMatchSnapshot();
  });

  test('when source defines url prefix and suffix', async () => {
    const docfy = new Docfy();
    const result = await docfy.run([
      {
        root,
        urlPrefix: 'docs',
        urlSchema: 'manual',
        urlSuffix: '.html',
        pattern: '**/*.md',
      },
    ]);

    const urls = [];
    result.content.forEach(page => {
      urls.push([page.source, page.meta.url]);
    });

    expect(urls).toMatchSnapshot();
  });
});
