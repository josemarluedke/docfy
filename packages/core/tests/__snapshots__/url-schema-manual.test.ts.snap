// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URLs should be correctly generated when urlSchema is set to manual base base, no url prefix, no suffix 1`] = `
[
  [
    "README.md",
    "/",
  ],
  [
    "docs/about.md",
    "/about",
  ],
  [
    "docs/custom-url.md",
    "/test-custom/url",
  ],
  [
    "docs/how-it-works.md",
    "/how-it-works",
  ],
  [
    "docs/html.md",
    "/html",
  ],
  [
    "docs/math.md",
    "/math",
  ],
  [
    "docs/static-asset.md",
    "/static-asset",
  ],
  [
    "docs/table.md",
    "/table",
  ],
  [
    "packages/package1/components/button.md",
    "/category1/components/button",
  ],
  [
    "packages/package1/docs/intro.md",
    "/category1/intro",
  ],
  [
    "packages/package2/docs/overview.md",
    "/category2/overview",
  ],
  [
    "packages/package2/docs/styles.md",
    "/category2/styles",
  ],
  [
    "packages/package1/components/form/index.md",
    "/category1/components/form",
  ],
  [
    "packages/package1/components/form/something-else.md",
    "/category1/components/something-else",
  ],
]
`;

exports[`URLs should be correctly generated when urlSchema is set to manual when source defines url prefix and suffix 1`] = `
[
  [
    "README.md",
    "/docs/",
  ],
  [
    "docs/about.md",
    "/docs/about.html",
  ],
  [
    "docs/custom-url.md",
    "/docs/test-custom/url.html",
  ],
  [
    "docs/how-it-works.md",
    "/docs/how-it-works.html",
  ],
  [
    "docs/html.md",
    "/docs/html.html",
  ],
  [
    "docs/math.md",
    "/docs/math.html",
  ],
  [
    "docs/static-asset.md",
    "/docs/static-asset.html",
  ],
  [
    "docs/table.md",
    "/docs/table.html",
  ],
  [
    "packages/package1/components/button.md",
    "/docs/category1/components/button.html",
  ],
  [
    "packages/package1/docs/intro.md",
    "/docs/category1/intro.html",
  ],
  [
    "packages/package2/docs/overview.md",
    "/docs/category2/overview.html",
  ],
  [
    "packages/package2/docs/styles.md",
    "/docs/category2/styles.html",
  ],
  [
    "packages/package1/components/form/index.md",
    "/docs/category1/components/form.html",
  ],
  [
    "packages/package1/components/form/something-else.md",
    "/docs/category1/components/something-else.html",
  ],
]
`;
